aioircd.py

```diff
@@ -247,11 +309,16 @@ async def read_lines(self):
                 raise MemoryError("Message exceed 1024 bytes")
 
             for line in lines:
-                yield line.decode()
+                try:
+                    line = line.decode('utf-8')
+                except UnicodeDecodeError:
+                    line = line.decode('latin-1')
+                yield line
+
+        if graceful_close:
+            self.writer.write_eof()
+            await self.writer.drain()
```

Comment:
La suite du très très mal à la tête tout ça

Reply:
Je note, si on utilise `terminate()` en dehors d'un `shutdown` alors on va `write_eof` 2x, on doit blinder la condition `if graceful_close` => `if graceful_close and not self.writer.is_closing()`